src/libServer/LookupServer.cpp

```diff
@@ -939,6 +1051,70 @@ string LookupServer::GetEthCall(const Json::Value& _json) {
   return result;
 }
 
+std::string LookupServer::GetWeb3ClientVersion() {
+  LOG_MARKER();
+  return "to do implement web3 version string";
+}
+
+string LookupServer::GetWeb3Sha3(const Json::Value& _json) {
+  LOG_MARKER();
+
+  const auto str{_json.asString()};
+  LOG_GENERAL(DEBUG, "GetWeb3Sha3 on:" << str);
+
+  return POW::BlockhashToHexString(ethash::keccak256(
+      reinterpret_cast<const uint8_t*>(str.data()), str.size()));
+}
+
+Json::Value LookupServer::GetEthMining() {
+  LOG_MARKER();
+  // @todo : the mining state a could be retrieved from the WorkServer if it can
+  // provide the exact state of mining.
+  return Json::Value(false);
+}
+
+std::string LookupServer::GetEthCoinbase() {
+  LOG_MARKER();
+  return "";
+}
+
+std::string LookupServer::GetNetVersion() {
+  LOG_MARKER();
+  return "";
+}
+
+Json::Value LookupServer::GetNetListening() {
+  LOG_MARKER();
+  return Json::Value(false);
+}
+
+std::string LookupServer::GetNetPeerCount() {
+  LOG_MARKER();
+  return "0x0";
+}
+
+std::string LookupServer::GetProtocolVersion() {
+  LOG_MARKER();
+  return "";
+}
+
+std::string LookupServer::GetEthChainId() {
+  LOG_MARKER();
+  return "0x0";
```

Comment:
My proposal is to use 0x8001 for Mainnet, or 0x814d for Tesnet.  See https://zilliqa-jira.atlassian.net/browse/ZIL-4668 for more details. Also https://dev.zilliqa.com/docs/apis/api-blockchain-get-network-id/

So we need to read constants.xml here, or in the constructor of the lookupServer, or, wherever Zilliqa chain_id can be found.